core/solver/gmres.cpp

```diff
@@ -156,7 +156,7 @@ void Gmres<ValueType>::apply_impl(const LinOp *b, LinOp *x) const
     while (true) {
         ++total_iter;
         this->template log<log::Logger::iteration_complete>(
-            this, total_iter, residual.get(), dense_x);
+            this, total_iter, residual.get(), dense_x, residual_norm.get());
         if (stop_criterion->update()
                 .num_iterations(total_iter)
                 .residual_norm(residual_norm.get())
```

Comment:
Maybe a good idea would be to pass all possible parameters here to the stopping criterion (such as the residual and solution) since they are available, this could be useful for using future stopping criterion.

Reply:
Good idea, update is now live.